R/module_tabs_with_filters.R

```diff
@@ -24,7 +24,36 @@
 #' @return A `tagList` of The main menu, place holders for filters and
 #'   place holders for the `teal` modules
 #'
+#' @examples
+#' # use non-exported function from teal
+#' example_modules <- getFromNamespace("example_modules", "teal")
```

Comment:
`example_datasets` doesn't exist and I'm agains creating internal object for purposes of the internal examples. Let's write "encapsulated" examples with full code. It is easier to control and to analyse.

Reply:
What do you mean?
```
You are using teal version 0.14.0.9039
> example_datasets
function() { # nolint
  dummy_cdisc_data <- example_cdisc_data()
  datasets <- teal_data_to_filtered_data(dummy_cdisc_data)
  list(
    "d2" = list(
      "d3" = list(
        "aaa1" = datasets,
        "aaa2" = datasets,
        "aaa3" = datasets
      ),
      "bbb" = datasets
    ),
    "ccc" = datasets
  )
}
<environment: namespace:teal>
```

Reply:
dummy functions were removed in pre-release-cleanup branch in following [commit](https://github.com/insightsengineering/teal/pull/1021/commits/ce0ba730fb1c1792381c0859c8e171a907588a42)

Reply:
done [92c4a63](https://github.com/insightsengineering/teal/pull/1026/commits/92c4a63eaa25230c2924dde378336e9481aa5b88)